test/e2e/sidecar_test.go

```diff
@@ -64,13 +65,20 @@ func (suite *SidecarTestSuite) AfterTest(suiteName, testName string) {
 
 // Sidecar runs a test with the agent as sidecar
 func (suite *SidecarTestSuite) TestSidecar() {
+
+	agentAsSideCarDeleted := false
+
 	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval}
 
 	jaegerInstanceName := "agent-as-sidecar"
 	j := getJaegerAgentAsSidecarDefinition(jaegerInstanceName, namespace)
+	defer func() {
+		if !agentAsSideCarDeleted {
```

Comment:
Why is this check needed? The variable is intialized and set in the middle of the tests. Isn't the deterministic at the end of the test?

Reply:
I extended this test to test the case addressed by this PR.

At some point in this test I create a new instance and delete the "agent-as-sidecar" Jaeger instance. If I don't do this check the test fails because it tries to delete an non-exist instance at the end.

Reply:
I also got confused by this, but I think I understand what's going on: if the test fails midway of the execution, this instance will be lingering around, which is why you need the conditional undeploy.

It might be better to change the `undeployJaegerInstance` to return a `bool` indicating whether it was able to find and remove the instance. In most cases, the caller wouldn't care about whether the Jaeger has been undeployed correctly, I suppose, but in the cases where it does care, it could just call `require.True(t, undeployJaegerInstance(j))`.

cc @kevinearls

Reply:
@jpkrohling agreed, I think we can change that to return a bool and let the caller decide if they care about failure.

Reply:
+1

Reply:
@jpkrohling @kevinearls @pavolloffay I'll do the suggested change but Is it ok to maintain this in a single test? or better to split?

Reply:
@rubenvp8510 I think it would be better to split.